src/backend/utils/fmgr/fmgr.c

```diff
@@ -869,10 +870,13 @@ fmgr_security_definer(PG_FUNCTION_ARGS)
 		 * We could be calling either a regular or a set-returning function,
 		 * so we have to test to see what finalize flag to use.
 		 */
-		pgstat_end_function_usage(&fcusage,
-								  (fcinfo->resultinfo == NULL ||
-								   !IsA(fcinfo->resultinfo, ReturnSetInfo) ||
-								   ((ReturnSetInfo *) fcinfo->resultinfo)->isDone != ExprMultipleResult));
+		
+		if (!pltsql_pgstat_function_check_hook || ((&fcusage)->fs != NULL
+		     && ((*pltsql_pgstat_function_check_hook)(fcinfo->flinfo->fn_oid))))
```

Comment:
We could only do the check for procedures, since with functions such a case would never arise.